frontend/src/components/GastoForm.js

```diff
@@ -6,39 +6,34 @@ class GastoForm extends React.Component {
     this.state = { gasto: props.gasto };
     this.handleChange = this.handleChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
-   
+    this.estadoInicial = this.estadoInicial.bind(this);
```

Comment:
Esto creo que no es necesario, porque no lo llamás directamente desde la vista sino a través de otro método que ya tiene el `this` bien puesto.